gtdynamics/utils/Trajectory.cpp

```diff
@@ -36,6 +36,21 @@ using std::vector;
 
 namespace gtdynamics {
 
+
+void Trajectory::addPhaseContactPoints(const Phase &phase) {
+  // Add unique PointOnLink objects to contact_points_
+  for (auto &&kv : boost::static_pointer_cast<const FootContactState>(phase.constraints())->contactPoints()) {
```

Comment:
I'm wondering why we cast all the time. I think maybe Phase should have a method called `footContactConstraintSpec`

Reply:
done